stl/lexer.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Lexing (tokenizing) a state transition (STL) spec."""
+
+# pylint: disable=g-doc-args
+# pylint: disable=g-docstring-missing-newline
+# pylint: disable=g-docstring-quotes
+# pylint: disable=g-no-space-after-docstring-summary
+# pylint: disable=g-short-docstring-punctuation
+# pylint: disable=g-short-docstring-space
+# pylint: disable=invalid-name
+# pylint: disable=unused-variable
+
+import logging
+import ply.lex  # pylint: disable=g-bad-import-order
+
+
+class StlSyntaxError(SyntaxError):
+  """Error for incorrect STL syntax."""
+
+
+class StlLexer(object):
+
+  def __init__(self, filename=None, **kwargs):
+    """Create a Lex lexer.
+
+    To pass this into a Ply Yacc parser, pass it in using the .lexer propert
+    of an StlLexer instance:
+      my_lexer = StlLexer()
+      my_parser = ply.yacc.parser(lexer=my_lexer.lexer)
+
+    Args:
+      filename: The filename string to use in any error messaging.
+      kwargs: Forwarded to ply.lex.lex.
+    """
+    self.filename = filename
```

Comment:
Done! Left the .lexer and .parser fields as is for now, since possibly will be using them as public